### PR TITLE
Pin revision for LayoutLMForQuestionAnswering and TFLayoutLMForQuestionAnswering tests

### PR DESCRIPTION
**Pull request by ankrgyl**
# What does this PR do?

The newly introduced tests for `LayoutLMForQuestionAnswering` and `TFLayoutLMForQuestionAnswering` broke due to a change to the weights in https://huggingface.co/impira/layoutlm-document-qa. To make sure a weights change does not break tests, I've pinned the revisions in these tests. I'll separately investigate the weights and debug if something is broken with them.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.

It was raised here: https://github.com/huggingface/transformers/pull/18407.

- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@NielsRogge @ydshieh

### CODE REVIEW

**Review by ydshieh — APPROVED** [review]
Thanks a lot, @ankrgyl !

File: src/transformers/models/layoutlm/modeling_layoutlm.py [hunk 1]
```diff
@@ -1285,7 +1285,9 @@ def forward(
         >>> import torch
 
         >>> tokenizer = AutoTokenizer.from_pretrained("impira/layoutlm-document-qa", add_prefix_space=True)
-        >>> model = LayoutLMForQuestionAnswering.from_pretrained("impira/layoutlm-document-qa")
+        >>> model = LayoutLMForQuestionAnswering.from_pretrained(
+        ...     "impira/layoutlm-document-qa", revision="1e3ebacae311132a7c4a8df2d350d7e4ffd6ff0a"
```
Comment (sgugger) [inline src/transformers/models/layoutlm/modeling_layoutlm.py:1289]: Note that you only need to put 7 characters for the commit hash, might be less intimating (and fit in one line) this way.

Comment (ankrgyl) [reply src/transformers/models/layoutlm/modeling_layoutlm.py]: Good point. Just updated.

**Review by sgugger — APPROVED** [review]
Perfect, thanks!

**Pull request merged**